### PR TITLE
Use DataSeriesCommon for thermal & wind 

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/libs/antares/study/parts/common/series.cpp [hunk 1]
```diff
@@ -52,5 +52,15 @@ uint64_t DataSeriesCommon::memoryUsage() const
...
     return timeSeries.memoryUsage();
 }
+
+double DataSeriesCommon::getValue(unsigned int hour, unsigned int year) const
+{
+    if (timeSeries.width == 1)
+        return timeSeries[0][hour];
+    if (timeseriesNumbers[0][year] < timeSeries.width)
+        return timeSeries[year][hour];
+    logs.warning() << "Invalid index";
```
Comment (a-zakir) [inline src/libs/antares/study/parts/common/series.cpp:62]: which one?

File: src/libs/antares/study/parts/load/container.cpp [hunk 2]
```diff
@@ -66,8 +66,10 @@ void Container::markAsModified() const
 
 uint64_t Container::memoryUsage() const
 {
-    return sizeof(Container) + ((!series) ? 0 : DataSeriesLoadMemoryUsage(series))
-           + ((!prepro) ? 0 : prepro->memoryUsage());
+    // TODO
```
Comment (a-zakir) [inline src/libs/antares/study/parts/load/container.cpp:69]: will this 'TODO'  be treated in this pr?

File: src/libs/antares/study/parts/solar/container.cpp [hunk 3]
```diff
@@ -68,8 +68,10 @@ void Container::markAsModified() const
 
 uint64_t Container::memoryUsage() const
 {
-    return sizeof(Container) + ((!series) ? 0 : DataSeriesSolarMemoryUsage(series))
-           + ((!prepro) ? 0 : prepro->memoryUsage());
+    // TODO
```
Comment (a-zakir) [inline src/libs/antares/study/parts/solar/container.cpp:71]: same

File: src/libs/antares/study/parts/wind/container.cpp [hunk 4]
```diff
@@ -37,44 +37,40 @@ namespace Data
 {
 namespace Wind
 {
-Container::Container() : prepro(nullptr), series(nullptr)
```
Comment (a-zakir) [inline src/libs/antares/study/parts/wind/container.cpp:40]: is there a reason to not 'in-class' intialize `prepro` ?

File: src/libs/antares/study/parts/wind/container.cpp [hunk 5]
```diff
@@ -37,44 +37,40 @@ namespace Data
...
     delete prepro;
-    delete series;
 }
 
 bool Container::forceReload(bool reload) const
 {
     bool ret = true;
-    if (series)
-        ret = series->forceReload(reload) and ret;
+    ret = series.forceReload(reload) and ret;
```
Comment (a-zakir) [inline src/libs/antares/study/parts/wind/container.cpp:52]: bool ret = series.forceReload(reload);

File: src/libs/antares/study/parts/wind/container.cpp [hunk 6]
```diff
@@ -37,44 +37,40 @@ namespace Data
...
     if (prepro)
         prepro->markAsModified();
 }
 
 uint64_t Container::memoryUsage() const
 {
-    return sizeof(Container) + ((!series) ? 0 : DataSeriesWindMemoryUsage(series))
-           + ((!prepro) ? 0 : prepro->memoryUsage());
+    return sizeof(Container) + series.memoryUsage()
+      + ((!prepro) ? 0 : prepro->memoryUsage());
```
Comment (a-zakir) [inline src/libs/antares/study/parts/wind/container.cpp:68]: complex ;)

File: src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp [hunk 7]
```diff
@@ -66,13 +66,6 @@ void ApplyRandomTSnumbers(const Study& study,
...
         }
-        // Wind
-        {
-            const Data::DataSeriesWind& data = *area.wind.series;
-            assert(year < data.timeseriesNumbers.height);
-            ptchro.Eolien
-              = (data.timeSeries.width != 1) ? (long)data.timeseriesNumbers[0][year] : 0; // zero-based
-        }
         // Renewable
         {
```
Comment (a-zakir) [inline src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp:70]: local scope instruction may be tidied in function
{} --> function?

File: src/solver/hydro/management/management.cpp [hunk 8]
```diff
@@ -401,7 +401,7 @@ void HydroManagement::prepareNetDemand(uint numSpace)
             if (parameters_.renewableGeneration.isAggregated())
             {
                 netdemand = +scratchpad.ts.load[ptchro.Consommation][hour]
-                            - scratchpad.ts.wind[ptchro.Eolien][hour] - scratchpad.miscGenSum[hour]
+                  - scratchpad.ts.wind.getValue(hour, year) - scratchpad.miscGenSum[hour]
```
Comment (a-zakir) [inline src/solver/hydro/management/management.cpp:404]: .getValue() --> what value?

File: src/solver/hydro/management/management.cpp [hunk 9]
```diff
@@ -532,9 +532,9 @@ void HydroManagement::makeVentilation(double* randomReservoirLevel,
     }
 
     if (parameters_.adequacy())
-        prepareNetDemand<Data::stdmAdequacy>(numSpace);
+        prepareNetDemand<Data::stdmAdequacy>(numSpace, y);
```
Comment (a-zakir) [inline src/solver/hydro/management/management.cpp:535]: i understand this is legacy code, but could you rename `y`

File: src/solver/simulation/simulation.h [hunk 10]
```diff
@@ -58,7 +58,8 @@ void SIM_RenseignementProblemeHebdo(const Antares::Data::Study& study,
                                     uint weekInTheYear,
                                     uint numSpace,
                                     const int,
-                                    const ALL_HYDRO_VENTILATION_RESULTS&);
+                                    const ALL_HYDRO_VENTILATION_RESULTS&,
```
Comment (a-zakir) [inline src/solver/simulation/simulation.h:61]: I think it's worth having a named parameter ;)

File: src/solver/variable/commons/wind.h [hunk 11]
```diff
@@ -197,10 +197,11 @@ class TimeSeriesValuesWind
         if (isRenewableGenerationAggregrated)
         {
             // The current wind time-series
-            (void)::memcpy(pValuesForTheCurrentYear[numSpace].hour,
-                           pArea->wind.series->timeSeries
-                             .entry[NumeroChroniquesTireesParPays[numSpace][pArea->index].Eolien],
-                           sizeof(double) * pArea->wind.series->timeSeries.height);
+            // TODO
```
Comment (a-zakir) [inline src/solver/variable/commons/wind.h:200]: TODO or not?

**Comment by flomnes** [comment]
I closed this PR. Will have to organize this a bit more, sorry.

**Pull request closed without merging**